File: app/src/main/java/closer/vlllage/com/closer/ui/ScrollStopperConstraintLayout.java
package closer.vlllage.com.closer.ui;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.constraint.ConstraintLayout;
import android.util.AttributeSet;
import android.view.MotionEvent;

public class ScrollStopperConstraintLayout extends ConstraintLayout {
    public ScrollStopperConstraintLayout(Context context) {
        super(context);
    }

    public ScrollStopperConstraintLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ScrollStopperConstraintLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onInterceptTouchEvent(@NonNull MotionEvent event) {
        if (getParent() != null) {
            getParent().requestDisallowInterceptTouchEvent(true);
        }

        return false;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/ui/ScrollStopperConstraintLayout.kt
package closer.vlllage.com.closer.ui

import android.content.Context
import androidx.constraintlayout.widget.ConstraintLayout
import android.util.AttributeSet
import android.view.MotionEvent

class ScrollStopperConstraintLayout : ConstraintLayout {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        if (parent != null) {
            parent.requestDisallowInterceptTouchEvent(true)
        }

        return false
    }
}
